Orange/data/pandas_compat.py

```diff
@@ -0,0 +1,81 @@
+"""Pandas DataFrame↔Table conversion helpers"""
+import numpy as np
+import pandas as pd
+from pandas.api.types import (
+    is_categorical_dtype, is_object_dtype,
+    is_datetime64_any_dtype, is_numeric_dtype,
+)
+
+from Orange.data import (
+    Table, Domain, DiscreteVariable, StringVariable, TimeVariable,
+    ContinuousVariable,
+)
+
+__all__ = ['table_from_frame']
+
+
+def table_from_frame(df, *, force_nominal=False):
+    """
+    Conver pandas.DataFrame to Orange.data.Table
+
+    Parameters
+    ----------
+    df : pandas.DataFrame
+    force_nominal : boolean
+        If True, interpret ALL string columns as nominal (DiscreteVariable)
+
+    Returns
+    -------
+    Table
+    """
+
+    def _is_discrete(s):
+        return (is_categorical_dtype(s) or
+                is_object_dtype(s) and (force_nominal or
+                                        s.nunique() < s.size**.666))
```

Comment:
I guess you do not want to use [is_discrete_values](https://github.com/biolab/orange3/blame/master/Orange/data/variable.py#L33) here as it is too complicated, but you could at least use the same constant :)

Reply:
👍

It's the exact same [universal] constant, just computed with more precision. 😛